src/util/set.js

```diff
@@ -1,43 +1,30 @@
-var toString = Object.prototype.toString
-var isDate = obj => toString.call(obj) === '[object Date]'
-
-module.exports =  class BadSet {
+module.exports = class BadSet {
 
   constructor(){
-    this._array = []
-    this.length = 0
+    this._map = {}
```

Comment:
you'd get some amount of improvement from using `Object.create(null)` here as well, since engines optimize those objects as hash lookups